SCClassLibrary/Common/GUI/PlusGUI/Control/ServerPlusGUI.sc

```diff
@@ -1,3 +1,23 @@
+ServerTreeView {
+	*new { |aServer, bounds=(Rect(128, 64, 400, 400)), interval=0.5|
+		var onClose, window;
+		window = Window.new(name.asString, bounds, scroll:true).front;
+		onClose = aServer.plotTreeView(interval, window.view, { defer {window.close}; });
+		window.onClose = {
+			onClose.value;
+		};
+		^window.view.hasHorizontalScroller_(false).background_(Color.grey(0.9));
+	}
+
+	*closeAll {
+		Window.allWindows.do{ |win|
+		if (win.name.debug == "ServerTreeView") { win.close }
+		}
+	}
+
+	close { this.close }
```

Comment:
hm – sure ? this will go into a loop.

Reply:
The first PR did not loop because it returned `Window`. However, I did it wrong, as you pointed out above. While preparing the second commit, I ran into a loop once and fixed it by rewriting all the code to move the `plotTreeView'.
https://github.com/supercollider/supercollider/pull/6294/commits/1bc7706c859daafa68f26caf704e501f8ad8c5d4